### PR TITLE
Fix store issues with SOS-5

### PR DESCRIPTION
**Pull request by narategithub**

### CODE REVIEW

File: ldms/src/store/slurm/store_slurm.c [hunk 1]
```diff
@@ -564,7 +566,9 @@ _open_store(struct sos_instance *si, ldms_set_t set)
 	}
 
  add_schema:
+	uuid_generate(schema_uuid);
 	slurm_schema_template.name = si->schema_name;
+	slurm_schema_template.uuid = (char*)schema_uuid;
```
Comment (tom95858) [inline ldms/src/store/slurm/store_slurm.c:571]: why is this cast necessary?

File: ldms/src/store/store_app/store_app.c [hunk 2]
```diff
@@ -332,6 +334,8 @@ __get_sos_schema(store_app_cont_t cont, const char *name,
 	if (!schema) {
 		/* Create a new schema for 'name' */
 		struct sos_schema_template *tmp = &app_schema_template;
+		uuid_generate_random(uuid);
```
Comment (tom95858) [inline ldms/src/store/store_app/store_app.c:337]: If we're generating random uuid, then it's not really useful for ensuring that schema across containers and partitions are the same. If we have multiple aggregators storing data for the same schema, they will have different UUID which means that attempting to combine these partitions later into a single container for analysis will fail...

**Review by tom95858 — COMMENTED** [review]
I think we need to have a discussion about how to handle this. The PAPI sampler comes to mind as a particular challenge.

File: ldms/src/store/slurm/store_slurm.c (hunk 1 above)
Comment (narategithub) [reply ldms/src/store/slurm/store_slurm.c:571]: Without the casting, GCC warned about "converts between pointers to integer types with different sign".

File: ldms/src/store/store_app/store_app.c (hunk 2 above)
Comment (narategithub) [reply ldms/src/store/store_app/store_app.c:337]: You're right. This should be the LDMS schema digest, and this patch must be on top of the LDMS schema digest patch. I'll close this pull request put this change on top of the schema digest commit.

**Comment by narategithub** [comment]
The `sos_schema.uuid` should be the LDMS schema digest. This PR is closed and the commit will be a part of #858.

**Pull request closed without merging**